### PR TITLE
Add metadata to action to allow extending the library

### PR DESCRIPTION
**Pull request by hogiyogi597**
**Description**

I have really enjoyed working with LetMe but I feel that some of what I'd _like_ to do is not possible without either updating the library to handle each case (doesn't make sense to do so) or by allowing library consumers the ability to attach metadata to their actions.

With a `metadata` field, there is no limit to how extensible the library can be!

**Checklist**

- [x] I added tests that cover my proposed changes.
- [x] I updated the documentation related to my changes (if applicable).

### CODE REVIEW

**Comment by woylie** [comment]
Thanks for the PR! This is useful, but I'd like to change the syntax a bit.

With this, we risk collisions, should we ever decide to add more options to the `action` macro.

```elixir
    action :disable, gql_exclude: true do
      allow role: :admin
    end
```

What do you think of this instead?

```elixir
    action :disable do
      allow role: :admin
      metadata gql_exclude: true
    end
```

This probably also reads more easily if you have more or longer meta data.

```elixir
    action :disable do
      allow role: :admin
      metadata [
        gql_exclude: true,
        doc_ja: "指定されたユーザーに対して、指定された機能を無効にします。"
      ]
    end
```

Or alternatively, we can enforce a keyword list for metadata and change the syntax to:

```elixir
    action :disable do
      allow role: :admin
      metadata :gql_exclude, true
      metadata :doc_ja, "指定されたユーザーに対して、指定された機能を無効にします。"
    end
```

What do you think?

**Comment by hogiyogi597** [comment]
Yeah! That is a great suggestion! I'll see what I can come up with!

**Comment by hogiyogi597** [comment]
Took a stab at it, let me (pun intended) know what you think! Happy to change whatever if you see more edits you'd like, @woylie !

**Comment by woylie** [comment]
Thanks, but I think I prefer the second syntax.

```elixir
metadata :gql_exclude, true
metadata :doc_ja, "指定されたユーザーに対して、指定された機能を無効にします。"
```

So here, the metadata macro would need to take two arguments. The resulting metadata value should be `[gql_exclude: true, doc_ja: "指定されたユーザーに対して、指定された機能を無効にします。"]`. Do you think you can change it this way?

**Comment by hogiyogi597** [comment]
Thanks for your response! In the case you are describing, you would only ever be able to write metadata in that form, correct? 

I think the way that it currently is, you can input metadata in a similar form (`metadata gql_exclude: true`) but also are not limited to inputting a keyword list of any size, right?

Is there a particular benefit in having it be the 2 argument form and preventing a keyword list? Happy to change it, just looking to understand better

**Comment by woylie** [comment]
It would be the only way to write it, but you could use the macro multiple times, so you can also have keyword lists of any size. I just think it reads nicer, kind of like the config macro.

**Comment by hogiyogi597** [comment]
Ok! I'll make the change :)

File: README.md [hunk 1]
```diff
@@ -241,6 +242,47 @@ iex> MyApp.Policy.list_rules(allow: {:role, :writer})
 ]
 ```
 
+Metadata can be defined on an `action` for use in extending LetMe's
+functionality that are out of scope for the library itself.
```
Comment (woylie) [inline README.md:246]:
```suggestion
You can also define metadata on an `action`, which can be useful to extend
the functionality of the library.
```

File: README.md [hunk 2]
```diff
@@ -241,6 +242,47 @@ iex> MyApp.Policy.list_rules(allow: {:role, :writer})
...
 
+You can also define metadata on an `action`, which can be useful to extend
+the functionality of the library.
+
+For example, you may want to expose some of your actions through your Absinthe
+GraphQL schema. Perhaps you wanted to exclude only certain actions. Using the
+`action`'s metadata, it becomes easier to do so.
+
+With `metadata`, there should be no limit to how the library can be extended
+in your own application.
```
Comment (woylie) [inline README.md:253]:
```suggestion
For example, you may want to expose some of your actions through your Absinthe
GraphQL schema, but exclude certain actions from the schema. You could solve
this by adding a `:gql_exclude` key to the metadata.
```

File: cheatsheets/rules.cheatmd [hunk 3]
```diff
@@ -271,3 +271,32 @@ object :article do
...
+    allow role: :editor
+    allow role: :writer
+    
+    metadata :gql_exclude, true
+  end
+end
+```
+
+### Metadata multiple
+
```
Comment (woylie) [inline cheatsheets/rules.cheatmd:290]: I think one example is enough here.

```suggestion
```

File: lib/let_me/policy.ex [hunk 4]
```diff
@@ -806,6 +808,32 @@ defmodule LetMe.Policy do
     end
   end
 
+  @doc """
+  Assigns metadata to the action in the form of a key value pair.
+  Keys should always be atoms.
```
Comment (woylie) [inline lib/let_me/policy.ex:813]:
```suggestion

  Keys should always be atoms.
```

**Comment by woylie** [comment]
Alright, I pushed some small documentation changes and an error in case the metadata key is not an atom. Thank you for the PR!

**Comment by hogiyogi597** [comment]
Thanks for working through it all with me! Excited to keep implementing our authorization with LetMe!

**Pull request merged**